libs/coin-evm/src/api/explorer/etherscan.ts

```diff
@@ -43,67 +54,272 @@ async function fetchWithRetries<T>(
 /**
  * Get all the last "normal" transactions (no tokens / NFTs)
  */
-export const getLastCoinOperations = makeLRUCache<
-  [currency: CryptoCurrency, address: string, accountId: string, fromBlock: number],
-  Operation[]
->(
-  async (currency, address, accountId, fromBlock) => {
-    const apiDomain = currency.ethereumLikeInfo?.explorer?.uri;
-    if (!apiDomain) {
-      return [];
-    }
+export const getLastCoinOperations = async (
+  currency: CryptoCurrency,
+  address: string,
+  accountId: string,
+  fromBlock: number,
+  toBlock?: number,
+): Promise<Operation[]> => {
+  const apiDomain = currency.ethereumLikeInfo?.explorer?.uri;
+  if (!apiDomain) {
```

Comment:
Can this actually happen ? Do we need to return an empty array in that case ?

Reply:
If the currency is misconfigured, yes it can happen. And that configuration will soon be at the user's level.